File: core/src/main/java/forpleuvoir/dhwu1a/core/event/friend/FriendNickChangedEvent.java
package forpleuvoir.dhwu1a.core.event.friend;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.common.data.FriendData;

import static forpleuvoir.dhwu1a.core.common.ApiKey.FROM;
import static forpleuvoir.dhwu1a.core.common.ApiKey.TO;

/**
 * 好友昵称改变
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.friend
 * <p>#class_name FriendNickChangedEvent
 * <p>#create_time 2021/7/3 21:06
 */
public class FriendNickChangedEvent extends FriendEvent {

    /**
     * 原昵称
     */
    @SerializedName(FROM)
    public final String from;
    /**
     * 修改后的昵称
     */
    @SerializedName(TO)
    public final String to;

    public FriendNickChangedEvent(FriendData friend, String from, String to) {
        super(FriendEventType.FriendNickChangedEvent, friend);
        this.from = from;
        this.to = to;
    }

    @Override
    public String toPlainText() {
        return String.format("%s[from:%s,to:%s]", type, from, to);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/friend/FriendNickChangedEvent.kt
package forpleuvoir.dhwu1a.core.event.friend

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.FROM
import forpleuvoir.dhwu1a.core.common.TO
import forpleuvoir.dhwu1a.core.common.data.FriendData

/**
 * 好友昵称改变
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.friend
 *
 * #class_name FriendNickChangedEvent
 *
 * #create_time 2021/7/3 21:06
 */
class FriendNickChangedEvent(
    friend: FriendData?,
    /**
     * 原昵称
     */
    @field:SerializedName(FROM) val from: String,
    /**
     * 修改后的昵称
     */
    @field:SerializedName(TO) val to: String
) : FriendEvent(FriendEventType.FriendNickChangedEvent, friend) {

    override fun callback() {
        friend?.let { bot.getFriend(friend.id)?.data?.nickname = to }
    }

    override fun toPlainText(): String {
        return String.format("%s[from:%s,to:%s]", type, from, to)
    }
}